### PR TITLE
general-pod-autoscaler support to set workers

### PR DESCRIPTION
**Pull request by jacky68147527**
Support to set workers, general-pod-autoscaler can be processed in parallel for workload autoscaling request.
args like this:
```
- args:
        - --election-resource-lock=endpoints
        - --tlscert=/root/cert.pem
        - --tlskey=/root/key.pem
        - --general-pod-autoscaler-workers=4
        - --v=4
        - --port=443
```

### CODE REVIEW

File: cmd/gpa/main.go [hunk 1]
```diff
@@ -117,6 +117,10 @@ func main() {
 		external_metrics.NewForConfigOrDie(kubeconfig),
 	)
 
+	workers := 1
+	if runConfig.GeneralPodAutoscalerWorkers > 0 {
+		workers = runConfig.GeneralPodAutoscalerWorkers
+	}
```
Comment (cwdsuzhou) [inline cmd/gpa/main.go]: move this to default.go

File: pkg/scaler/replica_calculator.go [hunk 2]
```diff
@@ -30,6 +30,7 @@ package scaler
 
 import (
 	"fmt"
+	"k8s.io/klog"
```
Comment (cwdsuzhou) [inline pkg/scaler/replica_calculator.go]: go fmt

File: pkg/scaler/replica_calculator.go [hunk 3]
```diff
@@ -142,7 +143,8 @@ func (c *ReplicaCalculator) GetResourceReplicas(currentReplicas int32, targetUti
 			metrics[podName] = metricsclient.PodMetric{Value: 0}
 		}
 	}
-
+	klog.V(4).Infof("GetResourceUtilizationRatio selector=%v, container=%v, resourceName=%v, resource=%v, "+
+		"targetUtilization=%v, usageRatio=%v, utilization=%v, rawUtilization=%v, rebalanceIgnored=%v", selector, container, resource, requests, targetUtilization, usageRatio, utilization, rawUtilization, rebalanceIgnored)
```
Comment (cwdsuzhou) [inline pkg/scaler/replica_calculator.go]: If not necessary, remove this.

File: .gitignore [hunk 4]
```diff
@@ -135,4 +135,7 @@ zz_generated.openapi.go
 # ignore binary
 /bin/
 
-./idea/
\ No newline at end of file
+./idea/
+
+# ignore go.sum
+go.sum
```
Comment (cwdsuzhou) [inline .gitignore]: do not ignore, thanks

File: pkg/scaler/general.go [hunk 5]
```diff
@@ -100,11 +100,16 @@ type GeneralController struct {
 	// Latest unstabilized recommendations for each autoscaler.
 	recommendations map[string][]timestampedRecommendation
 
+	// Multi goroutine read and write recommendations may unsafe.
+	recommendationsLock sync.RWMutex
+
 	// Latest autoscaler events
 	scaleUpEvents   map[string][]timestampedScaleEvent
 	scaleDownEvents map[string][]timestampedScaleEvent
```
Comment (cwdsuzhou) [inline pkg/scaler/general.go:109]: IMO, logics about events should also add lock

**Comment by jacky68147527** [comment]
Resubmit PR.

**Pull request closed without merging**